===================================================================
--- specs\javacard\framework\APDU.jml
+++ specs\javacard\framework\APDU.jml
@@ -18,6 +18,8 @@
 
 public final class APDU {
 
+	boolean SOMEMETHOD_FOR_TEST_123 (int i);
+
 	private static APDU _instance;
 
 	private static byte _tprotocol;

===================================================================
--- src\javacard\framework\APDU.java
+++ src\javacard\framework\APDU.java
@@ -14,7 +14,10 @@
 
 public final class APDU {
 
-    private /*@spec_public*/ short _Lc;
+    boolean SOMEMETHOD_FOR_TEST_123(int i){
+	return i == 0;
+};
+private /*@spec_public*/ short _Lc;
 
     private /*@spec_public*/ short _Lr;
 
