===================================================================
--- specs\javacard\framework\APDU.jml
+++ specs\javacard\framework\APDU.jml
@@ -173,7 +173,7 @@
 	ensures ! ( getProtocol() == PROTOCOL_T0 ) || ( \result == 258 );
 	ensures ! ( getProtocol() == PROTOCOL_T1 ) || ( ( \result > 0 && \result <= BUFFER_LENGTH ) );
 	@*/
-	public static /*@ pure @*/ short getOutBlockSize ();
+	public static /*@ pure @*/ short getOutBlockSize (int SOME_NEW_PARAM);
 
 	/*@
 	public normal_behavior

===================================================================
--- src\javacard\framework\APDU.java
+++ src\javacard\framework\APDU.java
@@ -89,7 +89,7 @@
             return _t1inBlockSize;
     }
 
-    public static short getOutBlockSize() {
+    public static short getOutBlockSize(int SOME_NEW_PARAM) {
         if (_tprotocol == PROTOCOL_T0)
             return (short) 258;
         else
