===================================================================
--- src\javacard\framework\APDU.java
+++ src\javacard\framework\APDU.java
@@ -115,7 +115,7 @@
         if (_apduState >= STATE_OUTGOING) {
             APDUException.throwIt(APDUException.ILLEGAL_USE);
         }
-        _apduState = STATE_OUTGOING;
+
         if (_tprotocol == PROTOCOL_T0)
             return (short) 256;
         else
