===================================================================
--- specs\javacard\framework\Applet.jml
+++ specs\javacard\framework\Applet.jml
@@ -28,7 +28,7 @@
 	/*@
 	public exceptional_behavior requires true;
 	@*/
-	public static void install (byte[] bArray, short bOffset, byte bLength) throws ISOException;
+	public static void install (byte[] bArray, byte bLength) throws ISOException;
 
 	/*@
 	public behavior requires apdu != null;

===================================================================
--- src\javacard\framework\Applet.java
+++ src\javacard\framework\Applet.java
@@ -17,7 +17,7 @@
     protected Applet() {
     }
 
-    public static void install(byte[] bArray, short bOffset, byte bLength)
+    public static void install(byte[] bArray,  byte bLength)
             throws ISOException {
         ISOException.throwIt(ISO7816.SW_FUNC_NOT_SUPPORTED);
     }
